cmake/project_version.cmake

```diff
@@ -0,0 +1,28 @@
+include_guard()
+
+function(obtain_project_version resultVarVersion resultVarGitHash)
+    if (GIT_FOUND)
+        execute_process(
+                COMMAND ${GIT_EXECUTABLE} --git-dir=${CMAKE_CURRENT_SOURCE_DIR}/.git describe --abbrev=0 --tags
+                OUTPUT_VARIABLE VERSION_STRING
+                OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
+        execute_process(
+                COMMAND ${GIT_EXECUTABLE} rev-parse HEAD
+                RESULT_VARIABLE git_result
+                OUTPUT_VARIABLE GIT_HASH
+                OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
+        if (NOT git_result)
+            set(${resultVarGitHash} ${GIT_HASH} PARENT_SCOPE)
+            message(STATUS "Building git hash: ${GIT_HASH}")
+        endif()
```

Comment:
In dev environments, inside git repository, the version will be inferred from tags and git hash saved to the build.